Cryo_crate Patch 1 + Extras

Viole7 opened the pull request:
*Changes crates to only spawn the contents upon opening.
*Possible fix to icon issue.
*Unsealed crates located on inactive z-levels delete after some time.
*Adds, removes, and re-balances some seeds in rarity.dm.
*Makes bees and bee-keeping gear obtainable. 
*Minor improvements and alterations.

<!-- 
!!IMPORTANT!!
Changes must be reviewed and approved by a developer before being merged.

If a developer has approved your PR, any other developer may immediately merge it, or the same developer may merge it after 24 hours.

A developer may not merge their own PR without the approval of two other developers.

Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Viole7 committed:
Cryo_crate Patch 1

*Changes crates to only spawn the contents upon opening.
*Possible fix to icon issues.
*Unsealed crates located on inactive z-levels delete after some time.
*Minor improvements and alterations.

Andrew-Fall committed:
Merge branch 'dev' into cryocrate_patch_1

Andrew-Fall commented on code/game/objects/structures/crates_lockers/closets/cryo_crate.dm:
```diff
@@ -42,17 +42,12 @@
-			var/mob/cold_critt = new content_path(src)
-			if(cold_critt in SSmobs.mob_list)
-				STOP_PROCESSING(SSmobs, cold_critt)
+    
```
There's 4 spaces in a row in this area instead of a tab

Viole7 replied:
No clue how that got there, thanks for spotting it.

Viole7 committed:
Update cryo_crate.dm

Fixes spaces

PsyCommando commented:
@Viole7 Just wondering, but can we find bees for bee hives in those crates?

Viole7 commented:
> 
> 
> @Viole7 Just wondering, but can we find bees for bee hives in those crates?

Not currently, but I don't see why they couldn't be added. I'll add them tomorrow along with some minor changes I want to make. (Done)

Viole7 committed:
Update seed_packets.dm

Adds bad tobacco seed packet.

Viole7 committed:
Update rarity.dm

*Adds bee packet as a seed. (Other bee accessories are still unobtainable.)
*Adds bad tobacco
*Removes weeds
*Adds algae
*Moves a few items around for better balance.

Viole7 committed:
Update service_fabricator.dm

Adds bee-keeping gear to service fabricator.

Viole7 committed:
Merge branch 'dev' into cryocrate_patch_1

PsyCommando committed:
Merge branch 'dev' into cryocrate_patch_1

PsyCommando reviewed (approved):
Looks good to me! Appearence probably won't work, but we really need this change for tomorrow.

Viole7 commented:
> 
> 
> Looks good to me! Appearence probably won't work, but we really need this change for tomorrow.

I tried creating an entry in the appearance datum, the closed state worked fine but for some reason the open state wouldn't display. I used VV verify the state was being assigned correctly, manually setting the icon var caused it to display properly. This despite both states residing in the same .dmi file, so it shouldn't have mattered. Not exactly sure what was going on. I saw that other objects were setting closet_appearance to null and manually assigning the icons, so I went with that. Is it the server saving that's assigning the icon to the default in the appearance datum? If so, what exactly is doing that? I've never actually setup my test server to save, so I've only been testing this stuff without saves.
